src/main/java/org/rumbledb/expressions/CommaExpression.java

```diff
@@ -64,9 +64,17 @@ private boolean bypassCurrentExpressionForExecutionModeOperations() {
     @Override
     public void initHighestExecutionMode() {
         if (bypassCurrentExpressionForExecutionModeOperations()) {
```

Comment:
I think the bypass is a leftover from earlier times that should no longer be needed.